config/catalog.go

```diff
@@ -0,0 +1,99 @@
+package config
+
+import (
+	"path/filepath"
+
+	"github.com/gruntwork-io/go-commons/errors"
+	"github.com/gruntwork-io/go-commons/files"
+	"github.com/gruntwork-io/terragrunt/pkg/log"
+	"github.com/gruntwork-io/terragrunt/util"
+	"github.com/hashicorp/hcl/v2"
+	"github.com/hashicorp/hcl/v2/gohcl"
+	"github.com/hashicorp/hcl/v2/hclparse"
+	"github.com/sirupsen/logrus"
+)
+
+type Catalog struct {
+	URLs []string `hcl:"urls,attr"`
+}
+
+func NewCatalog() *Catalog {
+	return &Catalog{}
+}
+
+func (cfg *Catalog) ParseConfigFile(configPath string) error {
+	log.Debugf("Reading Terragrunt cfg file at %s", configPath)
+
+	configString, err := util.ReadFileAsString(configPath)
+	if err != nil {
+		return err
+	}
+
+	// Parse the HCL string into an AST body that can be decoded multiple times later without having to re-parse
+	parser := hclparse.NewParser()
+	hclFile, err := parseHcl(parser, configString, configPath)
```

Comment:
It looks like you're doing custom parsing of the TG config here... Why not using the existing code that already knows how to do it?

One issue here is that the TG config file supports all sorts of features, such as `locals`, functions, etc... And if any of those are used in the `catalog` block, it's not clear if this custom parsing will support them.

Reply:
Ah, you are right. Redone. Thanks!